Formula/ghc@8.8.rb

```diff
@@ -0,0 +1,124 @@
+require "language/haskell"
+
+class Ghc < Formula
+  include Language::Haskell::Cabal
+
+  desc "Glorious Glasgow Haskell Compilation System"
+  homepage "https://haskell.org/ghc/"
+  url "https://downloads.haskell.org/~ghc/8.8.3/ghc-8.8.3-src.tar.xz"
+  sha256 "e0dcc0aaf3e234c5978f29e6df62947e97720ab404ec0158343df211c5480f89"
+
+  bottle do
+    sha256 "efba70e7b2639df91b366fb54580b2de9cf8b774c367e4eeadf662bf8b6c7696" => :catalina
+    sha256 "6ce78524676be247d1729f219faea5fc0c91f084f48fce18d77d590d7f9d1a1b" => :mojave
+    sha256 "8724b8bd4409b1500ef78616a3edce4843c8c47e540605224458f6ff82dc213d" => :high_sierra
+  end
+
+  head do
+    url "https://gitlab.haskell.org/ghc/ghc.git", :branch => "ghc-8.8"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+
+    resource "cabal" do
+      url "https://hackage.haskell.org/package/cabal-install-3.0.0.0/cabal-install-3.0.0.0.tar.gz"
+      sha256 "a432a7853afe96c0fd80f434bd80274601331d8c46b628cd19a0d8e96212aaf1"
+    end
+  end
```

Comment:
```suggestion
  keg_only :versioned_formula
```

Versioned formula should not have a `head` block. They also need to be keg only.